libr/util/str.c

```diff
@@ -3579,6 +3650,37 @@ R_API const char *r_str_str_xy(const char *s, const char *word, const char *prev
 	return d;
 }
 
+R_API size_t r_charset_encode_str(ut8 *asciistr, ut8 *in, size_t len_input, RCharset *r_char) {
+	bool is_in_set;
+	size_t j, k;
+	memset(asciistr, 0, len_input);
+	ut8 * buff_hex = (ut8 *) malloc (len_input);
+
+	for (j = 0; j < len_input; j ++) {
+		is_in_set = true;
+		buff_hex[0] = in[j];
+		buff_hex[1] = '\0';
+
+		for (k = 0; search_from_hex (r_char->custom_charset, buff_hex) == NULL && k < (len_input-j); k ++) {
+			buff_hex[k] = in[j+k];
+			if (k >= (len_input-j)-1) {
+				is_in_set = false;
+			}
+		}
+		buff_hex[k+1] = '\0';
+
+		if (is_in_set == true) {
```

Comment:
if (!is_in_set) { break;}

no need for else then

Reply:
Do you prefer with:

```
if (!is_in_set) { break;}
strcat ((char *) asciistr, (char *) search_from_hex (r_char->custom_charset, buff_hex)->ch);
```

I am not sure that it will work.

Reply:
no, dont use strcat, ever. and i typed the statement from the phone, you should indent it properly

Reply:
The problem is that search_from_hex (r_char->custom_charset, buff_hex)->ch); may now return a char string as well as a single char. Do you want that I reimplement strcat with a for loop to handle everything?

Reply:
Yes i know that the "chars" (or runes) are multiple chars. but those multiple chars can even contain null bytes like in the case of utf16. and strcat is totally unrecommended posix api to use. because it have no control over the buffer sizes. better use r_str_ncat